dynalab/tasks/task_io.py

```diff
@@ -0,0 +1,333 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import hashlib
+import json
+import os
+import uuid
+
+import requests
+from ts.context import Context
+
+from dynalab.tasks.annotation_mock_data import annotation_mock_data_generators
+from dynalab.tasks.annotation_verifiers import annotation_verifiers
+from dynalab_cli.utils import SetupConfigHandler
+
+
+class TaskIO:
+    def __init__(self, task_code, task_info_path=None):
+
+        if task_info_path is not None:
+            self.task_info = TaskIO.get_json_from_path(task_info_path)
+        else:
+            paths_to_check = [
+                f"./.dynalab/{task_code}.json",
+                f"/home/model-server/code/{task_code}.json",
+            ]
+            for path in paths_to_check:
+                self.task_info = TaskIO.get_json_from_path(path)
+                if self.task_info is not None:
+                    break
+
+        if self.task_info is None:
+            raise RuntimeError(f"No task io found.")
+
+        self.task_code = task_code
+        self.initialize_inputs_and_targets()
```

Comment:
Do we always need to run these functions? With the abstractions maybe these don't need to be attributes anymore?

Reply:
The variables initialized by the `initialize_inputs_and_targets()` method are always used, regardless of whichever method in the `TaskIO` class is used. So I think that it's reasonable to keep those variables as class attributes.

We can abstract out the `initialize_mock_data` method though :)

Reply:
I don't think you need anything to do with mock data when a model is deployed? Only the I/O part matters in that case. I wonder if we could reduce the size of `task_io` in the actual model server by removing these as attributes.

Reply:
Cool I saw you've changed it in the latest revision! It took me to the previous commit at first. Thanks for fixing this!